### PR TITLE
Add error message when tool is ivoked but not restored

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/cli/issues/5019
- When tool was not restored we currently get stack trace or really not user friendly error - this is addressing this issue

cc: @livarcocc @piotrpMSFT @jgoshi @jonsequitur

### CODE REVIEW

File: test/dotnet.Tests/PackagedCommandTests.cs [hunk 1]
```diff
@@ -124,6 +124,20 @@ public void CanInvokeToolFromDirectDependenciesIfPackageNameDifferentFromToolNam
...
+        {
+            var testInstance = TestAssets.Get("NonRestoredTestProjects", "AppWithNonExistingToolDependency")
+                .CreateInstance()
+                .WithSourceFiles();
+
+            new TestCommand("dotnet")
+                .WithWorkingDirectory(testInstance.Root)
+                .ExecuteWithCapturedOutput("nonexistingtool")
+                .Should().Fail()
+                    .And.HaveStdErrContaining("Version for package `dotnet-nonexistingtool` could not be resolved.");
```
Comment (livarcocc) [inline test/dotnet.Tests/PackagedCommandTests.cs]: File a issue for using a better error message when we can change those again.

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs [hunk 2]
```diff
@@ -75,6 +75,11 @@ private IEnumerable<NuGetVersion> GetAvailableToolVersions(string packageId)
             var availableVersions = new List<NuGetVersion>();
 
             var toolBase = GetBaseToolPath(packageId);
+            if (!Directory.Exists(toolBase))
+            {
+                return new NuGetVersion[0];
```
Comment (krwq) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/ToolPathCalculator.cs]: Thanks - I knew we had pattern for that but couldn't remember what that was 😄

File: test/dotnet.Tests/PackagedCommandTests.cs (hunk 1 above)
Comment (krwq) [inline test/dotnet.Tests/PackagedCommandTests.cs]: https://github.com/dotnet/cli/issues/5087

**Comment by krwq** [comment]
@MattGertz @srivatsn - can I get approval for this bug? This is fixing error message when tool for project is not restored (currently there stack trace or really unpleasant message is being displayed)

**Pull request merged**